ui/src/plugins/epl-ets/components/ETS.js

```diff
@@ -57,6 +65,92 @@ class ETS extends Component {
     window.removeEventListener('resize', this.resize);
   }
 
+  componentWillReceiveProps({ view }) {
+    if (view.length === 2 && !this.state.tabTraceId) {
+      this.setState({ view: [view[0]], loader: false });
+    } else {
+      this.setState({ view: view, loader: false });
+    }
+  }
+
+  handleTabClick = ({ tabId, tabTraceId }) => {
+    this.setState({
```

Comment:
React `setState` is not guaranteed to be sync and can be sometimes scheduled in different ways. It doesn't happen often right now but with new versions of React it will be more common.
In general every time you're updating state based on previous state use `setState` function syntax instead of `this.state` syntax because `this` syntax is binding value and state can be applied later. With function syntax you're always guaranteed to have recent state.
```js
this.setState(state => ({
  view: [state.clickedNode, tabId],
  tabTraceId: tabTraceId
}));
```